plugin/pkg/up/up.go

```diff
@@ -5,17 +5,14 @@ package up
 import (
 	"sync"
 	"time"
-
-	"github.com/cenkalti/backoff/v4"
 )
 
 // Probe is used to run a single Func until it returns true (indicating a target is healthy). If an Func
 // is already in progress no new one will be added, i.e. there is always a maximum of 1 checks in flight.
```

Comment:
```suggestion
// is already in progress no new one will be added, i.e. there is always a maximum of 1 checks in flight.
//
// There is a tradeoff to be made in figuring out quickly that an upstream is healthy and not doing to much work 
// (sending queries) to find that out. Having some kind of exp. backoff here won't help much, because you don't won't
// to backoff too much. You then also need random queries to be perfomed every so often to quickly detect a working
// upstream. In the end we just send a query every 0.5 second to check the upstream. This hopefully strikes a balance
// between getting information about the upstream state quickly and not doing too much work. Note that 0.5s is still an 
// eternity in DNS, so we may actually want to shorten it.   
```